fireflower/targets.py

```diff
@@ -95,39 +94,38 @@ def __init__(self, path, compressed=True, kwargs_in=None, kwargs_out=None):
     @contextmanager
     def _open_writer(self):
         with self.open('w') as f:
-            if self.compressed:
-                with gzip.GzipFile(filename=os.path.basename(self.path),
-                                   fileobj=f) as gzfile:
-                    yield gzfile
-            else:
-                yield f
+            yield f
 
     @contextmanager
     def _open_reader(self):
-        with self.open('r') as rawf:
-            if self.compressed:
-                with tempfile.TemporaryFile(mode='rw+b') as tmpfile:
-                    # gzip doesn't cooperate with luigi's S3 fake file
-                    # object so we need to copy to tmp
-                    shutil.copyfileobj(rawf, tmpfile)
-                    tmpfile.seek(0)
-                    yield tmpfile
-            else:
-                yield rawf
+        with self.open('r') as f:
+            yield f
 
     def write_csv(self, df, **kwargs):
         if self.kwargs_out:
             kwargs = toolz.merge(self.kwargs_out, kwargs)
         with self._open_writer() as f:
-            df.to_csv(f, **kwargs)
+            if self.compressed:
+                with tempfile.NamedTemporaryFile(mode='w+b') as tmp_w:
+                    df.to_csv(tmp_w.name, compression='gzip', **kwargs)
+                    with open(tmp_w.name, 'rb') as tmp_r:
+                        bytes_ = tmp_r.read()
+                        encoded_bytes = base64.b64encode(bytes_)
```

Comment:
No, the S3 file is still encoded in UTF-8. We need to do the base64 encoding as an intermediate encoding because the raw gzip bytes are not UTF-8 encodable.